### PR TITLE
feat: Add core-data metrics test

### LINKED ISSUE #681: Implement automation tests for core-data existing metrics
# 🚀 Feature Request

### Relevant Package [**REQUIRED**]

 This feature request is for testing core-data

### Description [**REQUIRED**]
The metrics `EventsPersisted` and `ReadingsPersisted` are implemented in Kamakura.
The metrics related test can be started from these two.

### PR DESCRIPTION
**Pull request by cherrycl**
Add tests for redis and mqtt both messagebus on separate file.

Fix #681

Signed-off-by: Cherry Wang <cherry@iotechsys.com>

**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [ ] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?)
- [ ] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

### CODE REVIEW

File: TAF/testScenarios/integrationTest/UC_metrics/core_data_metrics_mqtt.robot [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    When Create multiple events
+    And Sleep  ${interval}
+    Then Metrics EventsPersisted Should Be Received
+    [Teardown]  Run keywords  Delete device by name ${device_name}
+                ...           AND  Set Telemetry Metrics/EventsPersisted=false For core-data On Consul
+
+DataMetricsMQTT002-Disable EventsPersisted And Verify Metrics isn't Publish to MessageBus
+    Given Run MQTT Subscriber Progress And Output  edgex/telemetry/core-data/EventsPersisted  Payload
+    And Set Test Variable  ${device_name}  events-persisted-false
+    Given Set Telemetry Metrics/EventsPersisted=false For core-data On Consul
```
Comment (jumpingliu) [inline TAF/testScenarios/integrationTest/UC_metrics/core_data_metrics_mqtt.robot:36]:
```suggestion
    And Set Telemetry Metrics/EventsPersisted=false For core-data On Consul
```

File: TAF/testScenarios/integrationTest/UC_metrics/core_data_metrics_mqtt.robot [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    When Create multiple events
+    And Sleep  ${interval}
+    Then Metrics ReadingsPersisted Should Be Received
+    [Teardown]  Run keywords  Delete device by name ${device_name}
+                ...      AND  Set Telemetry Metrics/ReadingsPersisted=false For core-data On Consul
+
+DataMetricsMQTT004-Disable ReadingsPersisted And Verify Metrics isn't Publish to MessageBus
+    Given Run MQTT Subscriber Progress And Output  edgex/telemetry/core-data/ReadingsPersisted  Payload
+    And Set Test Variable  ${device_name}  readings-persisted-false
+    Given Set Telemetry Metrics/ReadingsPersisted=false For core-data On Consul
```
Comment (jumpingliu) [inline TAF/testScenarios/integrationTest/UC_metrics/core_data_metrics_mqtt.robot:57]:
```suggestion
    And Set Telemetry Metrics/ReadingsPersisted=false For core-data On Consul
```

File: TAF/testScenarios/integrationTest/UC_metrics/core_data_metrics_redis.robot [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    When Create multiple events
+    And Sleep  ${interval}
+    Then Metrics EventsPersisted Should Be Received
+    [Teardown]  Run keywords  Delete device by name ${device_name}
+                ...           AND  Set Telemetry Metrics/EventsPersisted=false For core-data On Consul
+
+DataMetricsRedis002-Disable EventsPersisted And Verify Metrics isn't Publish to MessageBus
+    Given Run Redis Subscriber Progress And Output  edgex.telemetry.core-data.EventsPersisted  telemetry
+    And Set Test Variable  ${device_name}  events-persisted-false
+    Given Set Telemetry Metrics/EventsPersisted=false For core-data On Consul
```
Comment (jumpingliu) [inline TAF/testScenarios/integrationTest/UC_metrics/core_data_metrics_redis.robot:36]:
```suggestion
    And Set Telemetry Metrics/EventsPersisted=false For core-data On Consul
```

File: TAF/testScenarios/integrationTest/UC_metrics/core_data_metrics_redis.robot [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    When Create multiple events
+    And Sleep  ${interval}
+    Then Metrics ReadingsPersisted Should Be Received
+    [Teardown]  Run keywords  Delete device by name ${device_name}
+                ...      AND  Set Telemetry Metrics/ReadingsPersisted=false For core-data On Consul
+
+DataMetricsRedis004-Disable ReadingsPersisted And Verify Metrics isn't Publish to MessageBus
+    Given Run Redis Subscriber Progress And Output  edgex.telemetry.core-data.ReadingsPersisted  telemetry
+    And Set Test Variable  ${device_name}  readings-persisted-false
+    Given Set Telemetry Metrics/ReadingsPersisted=false For core-data On Consul
```
Comment (jumpingliu) [inline TAF/testScenarios/integrationTest/UC_metrics/core_data_metrics_redis.robot:57]:
```suggestion
    And Set Telemetry Metrics/ReadingsPersisted=false For core-data On Consul
```

File: TAF/testScenarios/integrationTest/UC_metrics/core_data_metrics_mqtt.robot (hunk 1 above)
Comment (cherrycl) [reply TAF/testScenarios/integrationTest/UC_metrics/core_data_metrics_mqtt.robot]: Fixed.

File: TAF/testScenarios/integrationTest/UC_metrics/core_data_metrics_mqtt.robot (hunk 2 above)
Comment (cherrycl) [reply TAF/testScenarios/integrationTest/UC_metrics/core_data_metrics_mqtt.robot]: Fixed

File: TAF/testScenarios/integrationTest/UC_metrics/core_data_metrics_redis.robot (hunk 4 above)
Comment (cherrycl) [reply TAF/testScenarios/integrationTest/UC_metrics/core_data_metrics_redis.robot]: Fixed

File: TAF/testScenarios/integrationTest/UC_metrics/core_data_metrics_redis.robot (hunk 3 above)
Comment (cherrycl) [reply TAF/testScenarios/integrationTest/UC_metrics/core_data_metrics_redis.robot]: Fixed

**Review by jumpingliu — APPROVED** [review]
LGTM

**Pull request merged**